src/arrayMethodJoin.js

```diff
@@ -4,8 +4,31 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let strJoin = '';
+
+    if (this.length === 0) {
+      return strJoin;
+    }
+
+    for (let value of this) {
+      if (this.length === 1) {
+        strJoin += value;
+
+        return strJoin;
+      }
+
+      if (value === undefined || value === null) {
+        value = '';
+        strJoin += `${separator}${value}`;
+      } else {
+        strJoin += `${separator}${value}`;
+      }
+    }
+
+    const strJoinFix = strJoin.slice(`${separator}`.length, strJoin.length);
```

Comment:
```suggestion
    const strJoinFix = strJoin.slice(`${separator}`.length, strJoin.length);
```
Let's not use in this task built-in array methods (`slice` in this case). Try to make it with base loops and `if`s.
Instead of this `slice` here, you can check in your loop if it's not the last element in the array (for these purposes better to use common `for` loop instead of `for-of`:
`if (i < this.length -1)`
And add a separator to your string, If it's the last element - do nothing